components/collector/tests/source_collectors/gitlab/test_source_up_to_dateness.py

```diff
@@ -46,7 +50,22 @@ async def test_source_up_to_dateness_folder(self):
             response, value=str(self.expected_age), landing_url="https://gitlab.com/project/blob/branch/file"
         )
 
-    async def test_landing_url_on_failure(self):
+    async def test_source_up_to_dateness_pipeline(self):
+        """Test that the age of a pipeline can be measured."""
+        self.set_source_parameter("file_path", "")
+        build_date = datetime.fromisoformat(self.gitlab_jobs_json[0]["created_at"].strip("Z")).date()
+        expected_age = (date.today() - build_date).days
+        with self.patched_client_session_head():
+            response = await self.collect(get_request_json_return_value=self.gitlab_jobs_json)
+        self.assert_measurement(response, value=str(expected_age), landing_url="https://gitlab/project/-/pipelines/1")
+
+    async def test_file_landing_url_on_failure(self):
+        """Test that the landing url is the API url when GitLab cannot be reached."""
```

Comment:
Where is the default value for `file_path` set?

Reply:
In the superclass of the test class.